api/api.go

```diff
@@ -115,16 +118,30 @@ func sendGithubRequestHelper(endpoint string, token string) (res *http.Response,
 	req.Header.Add("Authorization", "Bearer "+token)
 	req.Header.Add("X-GitHub-Api-Version", "2022-11-28")
 
-	res, err = http.DefaultClient.Do(req)
+	retry_count := 0
 
-	if err != nil {
-		err = fmt.Errorf("Failed to send HTTP request")
-		statusCode = 500 // Internal server error
-	} else if res.StatusCode != 200 {
-		statusCode = res.StatusCode // forward API error code
-		err = fmt.Errorf("GitHub request responded with error code %d", statusCode)
+	for {
+		res, err = http.DefaultClient.Do(req)
+		retry_count += 1
+
+		if err != nil {
```

Comment:
is status code ==202 counted as an error? maybe it err != nil should go last

Reply:
I don't think so? [RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-successful-2xx) says this about 2xx status codes
> The 2xx (Successful) class of status code indicates that the client's request was successfully received, understood, and accepted.

Based on that definition, there would not be an error. Also, I tested this branch before making the PR. When there was a status code 202, this code was able to loop and retry the request. So, that means that `err` must have been nil when there is a 202.

Reply:
sounds good